libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -822,13 +853,15 @@ internal object MapboxRouteLineUtils {
             val trafficColor = getRouteColorForCongestion(
                 trafficIdentifier,
                 isPrimaryRoute,
+                isInactive = false,
```

Comment:
I'm simplifying the setup by calculating all segment colors exactly as before (in the "active" form) and only swapping them for "inactive" later on when needed. I think we should refactor this setup to calculate the correct segment color upfront but that's going to results in a lot more changes.

Reply:
So you rely on inactive styling being applied after this invocation? It can work, although it's easy to mess it up in the future.
It also looks like we have legIndex data here so we can decide whether it's an active leg or not.

Reply:
I'm relying on the existing mechanism, that's how it works today as well.

I don't see active leg index data here, but even if we had it, I don't want to rebuild the whole draw state and all `RouteLineExpressionData` on each active leg index change. I assume that's why there's a separate function that replaces only congestion color after the generation of segments and their offsets is done. The fact that we attach color in here might be worth cleaning up so that we separately calculate segment offsets and separately apply color. That could be a part of a future refactor though.

Reply:
OK.